### DISCUSSION TITLE
Tagging Multiple Genres With lastgenre

### DISCUSSION THREAD

**Discussion by supermepsipax** [beetbox/beets#5255] [Q&A] [answered]
I am trying to add multiple genres for each album in my library using the lastgenre plugin. I am having trouble with different file formats and getting a consistent output, with my current solution only working on mp3s and not on m4a or flac.

This is my config for the lastgenre plugin portion, the `\0` will be pythonized as a null character whenever it is used as a string and this works for any id3v2.4 tag.

```yaml
lastgenre:
    count: 3
    separator: "\0"
```
The issue is with flac or mp4/aac as it either drops all but the first genre, or concatenates them together into one large mutant genre when inspected with Picard, MediaInfo, and my media server/music players. I think with these formats there should be multiple instances of the genre field, like genre=Rock, genre=Britpop

Does anyone know of a workaround for this? I don't know if this is too picky of a request for a full feature request. I'm thinking of just defining my own separator, running lastgenre, and then making a custom script that will check the file types and separate and rewrite the tags based on the filetype.

I guess my other alternative is just individually tagging tracks with 1 genre and I will get something more in line with my expectations.

**Comment by JOJ0** [comment]
Hmm, there is multi-value-fields feature in Beets already but that is not implemented for the genre field yet. As a workaround, I can only suggest you try to work with a simple separator of `, ` and see how that goes. Other software/players will still see a "long/mutant" string, yes, sorry :-/

**Comment by alxtrnr** [comment]
> As a workaround, I can only suggest you try to work with a simple separator of ,  and see how that goes. 

That works for me with a count of three.

**Comment by opapanik** [comment]
Also run into this problem recently. Just started using beets and lastgenre on my music library consisting of flac files, then discovered that all my multi genre tags on each file where reduced to just one tag (multiple genres yes but on one tag with delimiter usage) :(
Would be nice if multiple genre tags could be supported in the future.

**Comment by JOJ0** [comment, marked as answer]
@opapanik multiple genre fields we have now in since a couple releases. Can we close this? :-)

**Reply by opapanik** [reply]
Haven't checked (have moved on) but yes, close it.
